solc/CommandLineInterface.cpp

```diff
@@ -574,6 +585,7 @@ bool CommandLineInterface::processInput()
 	};
 
 	m_compiler.reset(new CompilerStack(m_args.count(g_argAddStandard) > 0, fileReader));
+	auto scannerFromSourceName = [&](string const& _sourceName) -> solidity::Scanner const& { return m_compiler->scanner(_sourceName); };
```

Comment:
This is c++11 syntax for lambda functions. The `[&]` captures by reference and not by copy (`[=]` would be copy). So `scannerFromSourceName` is a function object that returns the scanner that belongs to the given source name.